bundles/org.openhab.binding.smartmeter/src/main/java/org/openhab/binding/smartmeter/internal/conformity/negate/NegateHandler.java

```diff
@@ -64,7 +68,15 @@ public static boolean shouldNegateState(String negateProperty,
      * @return Whether the given bit is set or not
      */
     public static boolean isNegateSet(String value, int negatePosition) {
-        long longValue = Long.parseLong(value);
+        NumberFormat format = NumberFormat.getInstance(Locale.getDefault());
```

Comment:
Are you sure that you want to use defaut locale which will be French for me for example, that means the decimal separator will be different depending on users locale ?

Reply:
It is assumed openHAB and the smartmeter exist in the same counhtry/locale. I thought to use the openHAB locale, imo that doesn't add much as in that case there might also be a mismatch in locale. This allready fixes many of the corner cases. 
Not sure if i can make a reliable detection of smartmeter locale easily. suggestions are welcome, otherwise i think it is better to merge it as is.

Reply:
I might lack some context here, but is it a matter of parsing a decimal number which can have either "." or "," as decimal separater? If yes, will there also be thousands separator, e.g. "1,000.5" vs. "1.000,5"? If not, I guess both "." and "," could be treated as a comma, i.e. one could be replaced with the other, and standard number parsing could be used without depending on locale?

Reply:
I'm not familiair with all possible output values (strings) of the supported smartmeter devices that this method relies on. Maybe @msteigenberger can confirm the value will now and in the future never be 1.000,0+

Reply:
It would be a problem if it could be an integer with a thousands separator: 1.000 / 1,000. Otherwise it would be simple even with a possible thousands separator as we could assume that the right-most "." or "," is the decimal separator and everything else can be stripped.

Reply:
Is the API documented? What does it say?

Reply:
There are many combinations (hardware / country, DSM specifications) so there does not seem to be one size fits all API-documentation. Up until now there are only a few combinationsw leading to this problem, for these affected there are some options here:

1. Assume system is set to the same (or similar non-conflicting) locale.
2. Assume openHAB locale is set to the same (or similar non-conflicting) locale.
3. Assume there are no thousand seperators, treating both `.` and `,` as seperator.

For this PR i chose option 1, but willing to adopt any other variant. If this will ever lead to new issues, a new PR is created easily.

Reply:
Does the used API really return different formats depending on the system locale ?
If not, just use appropriate locale

Reply:
And what would the appropiate locale be ?!

Reply:
The one in which the API returns value.
For example, if the API returns format 123.45 use Locale.ENGLISH ?

Reply:
As said [here](https://github.com/openhab/openhab-addons/pull/16183#discussion_r1462234843)

>There are many combinations (hardware / country, DSM specifications) so there does not seem to be one size fits all API-documentation

For example, here you can read the Dutch standard: https://www.netbeheernederland.nl/_upload/Files/Slimme_meter_15_a727fce1f1.pdf  on page 6.4 you can see taht a decimal point is used as seperator. The reported bug shows a value with a comma as seperator. 

Maybe it is better to close this PR and leave the bug as is, if there is too much doubts that this is a step forward.

Reply:
I'm still confused. When reading the issue, where does the "49,0" example come from? I saw it only mentioned here:
https://github.com/openhab/openhab-addons/issues/11029#issuecomment-1873494290, but not reported. My confusion is what lead you to the conclusion that "49,0" has to be handled?

I understand the issue that parsing as long will fail for both "49.0" and "49,0", but I don't understand where "49,0" came from in the first place, and it's not mentioned in that documentation either? So perhaps it would be sufficient to parse as double, or..?

Reply:
Hmm.  it was some weeks ago, can't recall exactly. Might be introduced with this test [here](https://github.com/openhab/openhab-addons/issues/11029#issuecomment-1873494290) as i was trying to prevent regressions and checked with other possible locales (comma is used in my locale) somehitng i allways tend to do.

So when i follow my own argument ;-)  i updated this PR, so that it derfaults to a `.` seperator as both dutch specsheet and the reported issue show a `.`